homeassistant/components/energyzero/services.py

```diff
@@ -75,12 +82,29 @@ def __serialize_prices(prices: Electricity | Gas) -> ServiceResponse:
     }
 
 
+async def __get_coordinator(
+    hass: HomeAssistant, call: ServiceCall
+) -> EnergyZeroDataUpdateCoordinator:
+    entry_id: str = call.data[ATTR_CONFIG_ENTRY]
+    entry: ConfigEntry | None = hass.config_entries.async_get_entry(entry_id)
+
+    if not entry:
+        raise HomeAssistantError(f"Invalid config entry: {entry_id}")
+    if entry.state != ConfigEntryState.LOADED:
+        raise HomeAssistantError(f"{entry.title} is not loaded")
+    if not (energyzero_domain_data := hass.data[DOMAIN].get(entry_id)):
```

Comment:
If the config entry is loaded we know that the key should existing in hass.data so I think we can remove this check.